### ISSUE TITLE
unnecessary_move lint, removing move with no effect

### ISSUE THREAD

**Issue by sukyoungjeong-furiosa** [rust-lang/rust-clippy#8751]
### What it does

clippy reports no warnings for unnecessary `move` keyword, such as:
```rust
let closure = move || value_moved(v);
```
Here `move` is unnecessary since there's no reference capture.

If a lint for above is acceptable, we may also have some warnings for belows.

```rust
std::thread::spawn(move || { value_moved(v) }
```

or

```rust
tokio::spawn(async move || value_moved_async(v).await}).await
```

### Lint Name

unnecessary_move

### Category

style

### Advantage

- Brings minimalism.

### Example

```rust
move || value_moved(v);
```

Could be written as:

```rust
|| value_moved(v);
```

**Comment by Kinrany** [comment]
Duplicate of https://github.com/rust-lang/rust-clippy/issues/8751

**Comment by smoelius** [comment]
> Duplicate of [#8751](https://github.com/rust-lang/rust-clippy/issues/8751)

@Kinrany Did you mean to post this on a different issue? #11721 maybe?

**Comment by Kinrany** [comment]
🤦 Yes, thank you!
